Firmware/HomeButtonsArduino/src/state_machine.h

```diff
@@ -61,8 +61,15 @@ class StateMachine {
     std::visit([](auto statePtr) { statePtr->entry(); }, current_state_);
   }
 
-  void execute_once() {
-    std::visit([](auto statePtr) { statePtr->execute_once(); }, current_state_);
+  void loop() {
+    if (first_run_) {
+      first_run_ = false;
+      base_.info("Entering state %s::%s", name_,
+                 std::visit([](auto statePtr) { return statePtr->get_name(); },
+                            current_state_));
+      std::visit([](auto statePtr) { statePtr->entry(); }, current_state_);
+    }
```

Comment:
Maybe to avoid duplicate code, enter and exit states action could be extracted in functions:
```c++
    if (first_run_) {
      first_run_ = false;
      _enter_state(current_state_);
    }
```
with
```c++
    void _enter_state()
    {
      base_.info("Entering state %s::%s", name_,
                 std::visit([](auto statePtr) { return statePtr->get_name(); },
                            current_state_));
      std::visit([](auto statePtr) { statePtr->entry(); }, current_state_);
    }
```

(could do the same for `_exit_state` even if there is no benefits right now)